connection_test.go

```diff
@@ -120,6 +120,68 @@ func TestClient_Connect(t *testing.T) {
 	})
 }
 
+func TestClient_Reconnect(t *testing.T) {
+	t.Run("reconnect called on new (never connected) connection", func(t *testing.T) {
+		server, err := NewTestServer()
+		require.NoError(t, err)
+		defer server.Close()
+
+		// our client can connect to the server
+		c, err := connection.New(server.Addr, testSpec, readMessageLength, writeMessageLength)
+		require.NoError(t, err)
+
+		err = c.Reconnect()
+		require.NoError(t, err)
+		require.NoError(t, c.Close())
+	})
+
+	t.Run("reconnect called on closed connection", func(t *testing.T) {
```

Comment:
let's add one more test when the connection was closed by the server and we call `Reconnect` (without calling `Close()`. I guess this is how we are going to use inside brand-gateway.

Here is how we can ask servier to close the connection: https://github.com/moov-io/iso8583-connection/blob/28c380de5cddac0ca6440280de32c2a4d8dd3579/connection_test.go#L696

Reply:
Are you referring to triggering `<-done` before the goroutine reaches it?